File: desafio-02/LaBatata101/python/desafio02.py
"""
Calculate all prime numbers from 1 to 10000
"""

def primes(limit):
    """ Calculate prime numbers from 2 to limit, using Sieve of Eratosthenes"""
    isprime = [True] * limit
    for i in range(2, limit):
        if isprime[i]:
            for factor in range(i**2, limit, i):
                isprime[factor] = False
    return isprime


if __name__ == '__main__':
    # Print prime numbers
    PRIME = primes(10000)
    for n in range(1, len(PRIME)):
        if PRIME[n]:
            print(n, end=' ')
    print('\n')

File: desafio-02/Doctor45dev/java/Desafio02.java
public class Desafio02 {
  public static void main(String[] args) {
    for (int i = 1; i < 10000; i++) {
      if (isPRIME(i)) {
        System.out.println(i);
      }
    }
  }

  private static boolean isPRIME(int n) {
    if (n <= 1)
      return false;
    if (n <= 3)
      return true;
    if (n % 2 == 0 || n % 3 == 0)
      return false;

    for (int i = 5; i * i <= n; i = i + 6) {
      if (n % i == 0 || n % (i + 2) == 0)
        return false;
    }
    return true;
  }
}
